### DISCUSSION TITLE
Drag&drop between windows on different workspaces

### DISCUSSION THREAD

**Discussion by ClmH** [linuxmint/discussions#180] [Ideas]
Hi,

Using Cinnamon, when I drag files from Nemo to the taskbar :

- over a button of a window on the same workspace : the window get the focus and I can drop the selected files in it

- over a button of a window from another workspace : the button start blinking red, and a "clone" of the button appears at the left of the window-list (blinking red too) ; no window gets the focus, it stays on the current workspace
(as I release the files I couldn't drop, the 2 buttons stay red (not blinking), and the "clone" does not respond to a right-click ; it stays red until I switch to another workspace)

My idea is to have the same behavior when the windows are on different workspaces: switch to the "destination" workspace then give focus to the "destination" window.

It already works like this on Xcfe and I use it a lot!

**Comment by ClmH** [comment]
I found a workaround (kind of): right-clic on the window button on the taskbar then select "Visible on every workspace".
Useful for attaching a file to an email, or that kind of thing...
But I still think it would be a more "natural" behavior to switch to the destination workspace and give focus to the destination window!

**Comment by mtwebster** [comment]
You can also switch workspaces while 'holding' a window using the arrow or number keys.

**Reply by ClmH** [reply]
Oh, nice trick, thanks!
